packages/theme_tailor/lib/src/util/recursive_import_locator.dart

```diff
@@ -26,82 +26,91 @@ import 'package:analyzer/dart/element/element.dart';
 import 'package:collection/collection.dart';
 
 extension FindAllAvailableTopLevelElements on LibraryElement {
+  bool isFromPackage(String packageName) {
+    return librarySource.fullName.startsWith('/$packageName/');
+  }
+
   /// Recursively loops at the import/export directives to know what is available
   /// in the library.
   ///
   /// This function does not guarantees that the elements returned are unique.
   /// It is possible for the same object to be present multiple times in the list.
-  Iterable<Element> allVisibleElements() {
-    final elements = libraryImports.imports().expand((import) => import.library._visibleElements({}, import.key));
-
-    return elements;
+  Iterable<Element> findAllAvailableTopLevelElements() {
+    return _findAllAvailableTopLevelElements(
+      {},
+      checkExports: false,
+      key: _LibraryKey(
+        hideStatements: {},
+        showStatements: {},
+        librarySource: librarySource.fullName,
+      ),
+    );
   }
 
-  Iterable<Element> _visibleElements(
-    Set<String> visitedPaths,
-    _LibraryKey key,
-  ) sync* {
+  Iterable<Element> _findAllAvailableTopLevelElements(
+    Set<_LibraryKey> visitedLibraryPaths, {
+    required bool checkExports,
+    required _LibraryKey key,
+  }) sync* {
     yield* topLevelElements;
 
-    for (final directive in libraryExports.exports()) {
-      if (!visitedPaths.add(key.toString() + directive.key.toString())) {
+    final librariesToCheck = checkExports
+        ? units.expand((e) => e.libraryExports).map(_LibraryDirectives.fromExport).nonNulls
+        : units.expand((e) => e.libraryImports).map(_LibraryDirectives.fromImport).nonNulls;
+
+    for (final directive in librariesToCheck) {
+      if (!visitedLibraryPaths.add(directive.key)) {
         continue;
       }
 
-      yield* directive.library._visibleElements(visitedPaths, directive.key).where(directive.elementIsExported);
+      yield* directive.library
+          ._findAllAvailableTopLevelElements(
+        visitedLibraryPaths,
+        checkExports: true,
+        key: directive.key,
+      )
+          .where(
+        (element) {
+          return (directive.showStatements.isEmpty && directive.hideStatements.isEmpty) ||
```

Comment:
This condition could be broke down into 2/3 fields and then returned combined
```
final a = ...
final b = ...
return a || b;
```

Reply:
I've used Freezed' implementation from 49b3b930eab79381570967e9f4cbb9f8cd1cc30a. Copyright on top of the file